### PR TITLE
[ BB2-1094 ] combining all together

### PR DESCRIPTION
**Pull request by nbragdon**

### CODE REVIEW

**Review by JFU-GIT — COMMENTED** [review]
Like the refactored axios error handling.

File: src/auth.ts [hunk 1]
```diff
@@ -113,31 +113,31 @@ function validateCallbackRequestQueryParams(
...
+    authData,
     callbackRequestCode,
     callbackRequestState,
     callbackRequestError
   );
 
-  const postData = generateTokenPostData(bb, AuthData, callbackRequestCode);
+  const postData = generateTokenPostData(bb, authData, callbackRequestCode);
 
   const resp = await axios.post(getAccessTokenUrl(bb), postData);
```
Comment (dtisza1) [inline src/auth.ts:142]: @nbragdon We should add the SDK headers to the post().

**Review by dtisza1 — APPROVED** [review]
@nbragdon This looks good to me!

Great work combining everything together and the improvements!!!

File: src/index.ts [hunk 2]
```diff
@@ -85,23 +101,79 @@ export default class BlueButton {
...
+  }
+
+  async getPatientData(
+    authToken: AuthorizationToken,
+    config: AxiosRequestConfig
+  ) {
+    return await getFhirResource(
+      FhirResourceType.Patient,
+      authToken,
+      this,
```
Comment (JFU-GIT) [inline src/index.ts:123]: in the scope of an async method in a class, "this" is not a reference to the class instance - that's what I ran into, searched internet and not finding anything though....

**Review by JFU-GIT — APPROVED** [review]
Looks good to me, if there were issues - could be resolved at integration time (when drop into the sample app)...

**Pull request merged**